src/Formats/FormatSettings.h

```diff
@@ -91,6 +92,8 @@ struct FormatSettings
         bool crlf_end_of_line = false;
         bool input_format_enum_as_number = false;
         bool input_format_arrays_as_nested_csv = false;
+        bool read_bool_as_uint8 = false;
+        Names input_field_names;
```

Comment:
If we use separate struct for HiveText, then `HiveTextRowInputFormat` use both `format_settings.csv` and `format_settings.hive_text`,  that is strange.

Reply:
The fact that we have `input_field_names` in `CSV` format and dont't use it in `CSV` is strange. The fact that the `HiveText` format uses `format_settings.csv` is not strange, because `HiveText` is implemented by `CSV` format.

Reply:
For example, `Template` or `Regexp` formas use format_settings.csv and it's ok, because they can use CSV deserialization

Reply:
ok